tests/functional/disaster-recovery/metro-dr/test_replace_cluster.py

```diff
@@ -0,0 +1,209 @@
+import logging
+import pytest
+import time
+
+from concurrent.futures import ThreadPoolExecutor
+from ocs_ci.framework import config
+from ocs_ci.ocs.acm.acm import AcmAddClusters
+from ocs_ci.ocs import constants
+
+from ocs_ci.framework.pytest_customization.marks import tier1, turquoise_squad
+from ocs_ci.helpers.dr_helpers import get_active_acm_index
+from ocs_ci.ocs.node import get_node_objs
+from ocs_ci.ocs.dr.dr_workload import validate_data_integrity
+from ocs_ci.helpers.dr_helpers_ui import (
+    check_cluster_status_on_acm_console,
+    failover_relocate_ui,
+)
+from ocs_ci.helpers.dr_helpers import (
+    replace_cluster,
+    enable_fence,
+    enable_unfence,
+    get_fence_state,
+    failover,
+    relocate,
+    set_current_primary_cluster_context,
+    get_current_primary_cluster_name,
+    get_current_secondary_cluster_name,
+    wait_for_all_resources_creation,
+    gracefully_reboot_ocp_nodes,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@turquoise_squad
+class TestReplaceCluster:
+    """
+    Test for Recovery of replacement cluster
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request, cnv_dr_workload):
+        """
+        Teardown function: If fenced, un-fence the cluster and reboot nodes
+        """
+
+        def finalizer():
+            if (
+                self.primary_cluster_name
+                and get_fence_state(self.primary_cluster_name) == "Fenced"
+            ):
+                enable_unfence(self.primary_cluster_name)
+                gracefully_reboot_ocp_nodes(
+                    self.primary_cluster_name, disable_eviction=True
+                )
+
+        request.addfinalizer(finalizer)
+
+    def test_replace_cluster(
+        self,
+        nodes_multicluster,
+        dr_workload,
+    ):
+        """
+        Tests to verify Recovery of replacement cluster
+        """
+
+        # Failover apps followed by unfence
+
+        acm_obj = AcmAddClusters()
+
+        # Deploy Subscription based application
+        sub = dr_workload(num_of_subscription=1, switch_ctx=get_active_acm_index())[0]
+        self.namespace = sub.workload_namespace
+        self.workload_type = sub.workload_type
+
+        """
+        # Deploy AppSet based application
```

Comment:
@avd-sagare Aren't we adding appset based workload?

Reply:
Yes we are adding, but 1st testing for subscription

Reply:
Added appset based workload

Reply:
Facing issue for sub + appset So raised issue. Now decided to go with sub type only